src/base_alloc/base_alloc_os_mem.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+*/
+
+#include <assert.h>
+#include <stdio.h>
+
+#include <base_alloc_os_mem_internal.h>
+
+#include "utils_common.h"
+
+// size of buffer for storing memory regions
+#define SIZE_OS_MEM_REGIONS_BUFFER 100
+
+typedef struct os_mem_region_t {
+    void *ptr;
+    size_t size;
+} os_mem_region_t;
+
+typedef struct os_mem_t {
+    os_mem_region_t region[SIZE_OS_MEM_REGIONS_BUFFER];
+    int index;
+} os_mem_t;
+
+static __TLS os_mem_t OS_mem;
+
+void *ba_os_alloc(size_t size) {
+    // assert(OS_mem.index < SIZE_OS_MEM_REGIONS_BUFFER - 1);
```

Comment:
why is this commneted out? we should have some check like this, right? (and probably return an error, even in release build)

Reply:
I am not sure if we should assert in this case, when we can allocate more memory, but we cannot only save this pointer in the array, so we will leak this memory ...

Reply:
Removed.
Done.